stablehlo/dialect/VhloTypes.td

```diff
@@ -86,6 +86,12 @@ def VHLO_FloatF8E4M3FNV1 : VHLO_TypeDef<"FloatF8E4M3FNV1", "f8E4M3FN_v1", "0.9.0
 // Corresponds to the 'f8E5M2' FloatType from the StableHLO spec.
 def VHLO_FloatF8E5M2V1 : VHLO_TypeDef<"FloatF8E5M2V1", "f8E5M2_v1", "0.9.0", "current">;
 
+// Corresponds to the 'f8E4M3FNUZ' FloatType from the StableHLO spec.
+def VHLO_FloatF8E4M3FNUZV1 : VHLO_TypeDef<"FloatF8E4M3FNUZV1", "f8E4M3FNUZ_v1", "0.9.0", "current">;
+
+// Corresponds to the 'f8E5M2FNUZ' FloatType from the StableHLO spec.
+def VHLO_FloatF8E5M2FNUZV1 : VHLO_TypeDef<"FloatF8E5M2FNUZV1", "f8E5M2FNUZ_v1", "0.9.0", "current">;
+
```

Comment:
A few notes on the compatibility side, since adding a new type is not forward compatible and requires a new version number:

- These should be `"0.10.0", "current"`
- `stablehlo/dialect/Version.h` function `getCurrentVersion` should get bumped to `(0,10,0)`.
- `stablehlo/tests/stablehlo_legalize_to_vhlo.mlir` needs a test (looks like you have this covered already)
- Add an invalid downgrade test. Unfortunately this is the first opset change since our public release so there isn't a dedicated test file yet, but look at [vhlo_to_version_downgrade_invalid.mlir](https://github.com/openxla/stablehlo/blob/6886b59f6cd4369674e7e3beff61301c145176e2/stablehlo/tests/vhlo_to_version_downgrade_invalid.mlir#L102) from before the public release cleanup for examples of what this test looked like for fp8_E5M2. The new test should specify `0.9.0` as the target version and error if a new FP8 type is used.

Reply:
Moved to https://github.com/openxla/stablehlo/pull/1379#discussion_r1159924774